### DISCUSSION TITLE
Nuclei failed to resolve CNAME domain name

### DISCUSSION THREAD

**Discussion by bit4woo** [projectdiscovery/nuclei#1545] [Q&A] [answered]
### Nuclei version:

Nuclei Engine 2.5.7 (latest)
### Steps To Reproduce:

summary: 
unclei failed to resolve CNAME domain. lead to PoC check failed.
at the sametime. A Record Domain and IP address works well for same target.

detail:

nuclei -t /Users/bit4woo/nuclei-templates/bit4woo/grafana-file-read-bypass-nginx.yaml -u http://win.yourCNameDomain.com:9200 -debug -elog error.log 

![image](https://user-images.githubusercontent.com/8793481/148322040-621bd081-0b7a-4291-b573-22e456d6d76d.png)

![image](https://user-images.githubusercontent.com/8793481/148322088-23382bcb-6689-447d-ad14-e86aee0b337a.png)

nuclei -t /Users/bit4woo/nuclei-templates/bit4woo/grafana-file-read-bypass-nginx.yaml -u http://crwyrf.cn.corp.ARecordDomain.com:9200 -elog error.log

![image](https://user-images.githubusercontent.com/8793481/148322248-81d485c8-e129-4af2-a0ef-c0a5bdca88d1.png)

 nuclei -t /Users/bit4woo/nuclei-templates/bit4woo/grafana-file-read-bypass-nginx.yaml -u http://10.12.72.202:9200 -elog error.log

![image](https://user-images.githubusercontent.com/8793481/148322338-f7b4ec69-b84e-47ec-9287-8bbfeec794f3.png)

**Comment by ehsandeep** [comment]
@bit4woo do you have any host to share to reproduce this scenario?

```console
host -t cname docs.hackerone.com

docs.hackerone.com is an alias for hacker0x01.github.io.
```

```console
nuclei -t technologies/tech-detect.yaml -u https://docs.hackerone.com -silent

[tech-detect:gatsby] [http] [info] https://docs.hackerone.com
[tech-detect:jsdelivr] [http] [info] https://docs.hackerone.com
[tech-detect:react] [http] [info] https://docs.hackerone.com
```

**Comment by bit4woo** [comment]
@ehsandeep thanks for reply. I think you should try a PoC which type is "network".

I can reproduce this scenario with your domain "docs.hackerone.com"

you can try with this PoC :  nuclei-templates/network/mongodb-detect.yaml

but something different this time : not only CNAME.   ALL CNAME, A Record,  IP  Address show same error.

Hope it's helpful for you.

![image](https://user-images.githubusercontent.com/8793481/148485402-27c968c0-e3ad-4ba2-ae37-89c7d6b426b1.png)

![image](https://user-images.githubusercontent.com/8793481/148485490-5dc3667e-3a02-419b-b873-a75685bc1480.png)

![image](https://user-images.githubusercontent.com/8793481/148485537-a2616000-d5d5-445e-83e6-60628df21d0d.png)

**Comment by ehsandeep** [comment, marked as answer]
@bit4woo both errors are expected as none of the input running `mongodb`, let me know if have example where you have service running and nuclei unable to connect it.
